dynamodb.go

```diff
@@ -0,0 +1,156 @@
+package limiters
+
+import (
+	"context"
+	"sync"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+	"github.com/pkg/errors"
+)
+
+// DynamoDBTableProperties are supplied to DynamoDB limiter backends.
+// This struct informs the backend what the name of the table is and what the names of the key fields are.
+type DynamoDBTableProperties struct {
+	//TableName is the name of the table.
+	TableName string
+	//PartitionKeyName is the name of the PartitionKey attribute.
+	PartitionKeyName string
+	//SortKeyName is the name of the SortKey attribute.
+	SortKeyName string
+	//SortKeyUsed indicates if a SortKey is present on the table.
+	SortKeyUsed bool
+	//TTLFieldName is the name of the attribute configured for TTL.
+	TTLFieldName string
+	//TTLUsed indicates if TTL is configured on the table.
+	TTLUsed bool
+}
+
+var tableCache = make(map[string]DynamoDBTableProperties)
+var lock = &sync.Mutex{}
```

Comment:
Using global variables is discouraged. 
I'd suggest removing them and instead make the client code implement the cache if needed.
Same goes for the mutex: let the client code control the concurrent execution behavior.

Reply:
removed